webapp/src/hooks/usePluginReduxSyncState.ts

```diff
@@ -0,0 +1,102 @@
+import {useEffect} from 'react';
+import {useDispatch} from 'react-redux';
+
+import pluginConstants from 'pluginConstants';
+
+import {resetGlobalModalState} from 'reducers/globalModal';
+import {toggleIsLinkedProjectListChanged, toggleShowLinkModal} from 'reducers/linkModal';
+import {toggleShowSubscribeModal} from 'reducers/subscribeModal';
+import {toggleShowTaskModal} from 'reducers/taskModal';
+import {getGlobalModalState, getLinkModalState, getSubscribeModalState, getCreateTaskModalState, getRhsState, getWebsocketEventState} from 'selectors';
+
+import usePluginApi from 'hooks/usePluginApi';
+import useApiRequestCompletionState from 'hooks/useApiRequestCompletionState';
+
+/**
+ * This hook intercepts the redux changes for the plugin to perform common actions like
+ * checking if the user's account is connected
+ * closing any open modal if the user is logged out from another client tab
+ * preventing opening a modal if the user's account is not connected
+ * fetching linked project list on the opening of RHS, on linking of a new project, etc
+ */
+function usePluginReduxSyncState() {
```

Comment:
```suggestion
function usePluginReduxState() {
```
I don't understand the meaning of `sync` here.

Reply:
actually here we are reading some redux states and based on that updating/modifying other global redux states of the plugin like modals and fetching the list of project list, etc
So, in a way, we are syncing the plugin's redux states internally, also this name was suggested by mickmister [here](https://github.com/mattermost/mattermost-plugin-azure-devops/pull/141#discussion_r1081985432)